cmd/juju/commands/debuglog.go

```diff
@@ -270,41 +281,76 @@ func isTerminal(f interface{}) bool {
 }
 
 // Run retrieves the debug log via the API.
-func (c *debugLogCommand) Run(ctx *cmd.Context) (err error) {
+func (c *debugLogCommand) Run(ctx *cmd.Context) error {
 	if c.tail {
 		c.params.NoTail = false
-	} else if c.notail {
+	} else if c.noTail {
 		c.params.NoTail = true
 	} else {
 		// Set the default tail option to true if the caller is
 		// using a terminal.
 		c.params.NoTail = !isTerminal(ctx.Stdout)
 	}
 
-	client, err := getDebugLogAPI(c)
-	if err != nil {
-		return err
-	}
-	defer client.Close()
-	messages, err := client.WatchDebugLog(c.params)
-	if err != nil {
-		return err
-	}
 	writer := ansiterm.NewWriter(ctx.Stdout)
 	if c.color {
 		writer.SetColorCapable(true)
 	}
-	for {
-		msg, ok := <-messages
-		if !ok {
-			break
-		}
-		c.writeLogRecord(writer, msg)
+
+	err := retry.Call(retry.CallArgs{
+		Func: func() error {
+			client, err := getDebugLogAPI(c)
+			if err != nil {
+				return err
+			}
+			defer client.Close()
+
+			messages, err := client.WatchDebugLog(c.params)
+			if err != nil {
+				return err
+			}
+
+			for {
+				msg, ok := <-messages
+				if !ok {
+					return ErrConnectionClosed
+				}
+				c.writeLogRecord(writer, msg)
+			}
+		},
+		IsFatalError: func(err error) bool {
+			if !c.retry {
+				return true
+			}
+			if errors.Is(err, ErrConnectionClosed) {
+				return false
+			}
+			return true
+		},
+		NotifyFunc: func(err error, attempt int) {
+			logger.Debugf("retrying to connect to debug log")
+		},
+		Attempts: -1,
+		Clock:    clock.WallClock,
+		Delay:    c.retryDelay,
+		Stop:     ctx.Done(),
+	})
+
+	// Ensure that any sentinel ErrConnectionClosed errors are not shown to the
+	// user. As this is a synthetic error that is used to signal that the
+	// connection is retried, we don't want to show this to the user.
+	if errors.Is(err, ErrConnectionClosed) {
+		return nil
 	}
 
-	return nil
+	// Unwrap the retry call error trace for all errors. We don't want to show
+	// that to the user as part of the error message.
+	return errors.Cause(err)
 }
 
+// Sentinel error used to signal that the connection is closed.
```

Comment:
Comment should begin with var name.
I also like `errors.ConstError` for these custom errors these days.